lms/static/scripts/frontend_apps/index.tsx

```diff
@@ -22,11 +22,10 @@ import type { ServiceMap } from './services';
  * those routes and reloading the page results in a 404.
  */
 function routeForAppMode(mode: AppMode): string {
-  if (mode === 'email-preferences') {
-    // For the email-preferences mode, since this app is not designed to be
-    // opened in an iframe, but as the main window frame, we want to use a route
-    // that matches the server-side one.
-    return '/email/preferences';
+  // For apps not designed to be opened in an iframe, but as the main window
+  // frame, we want to keep the server-side route.
+  if (['dashboard', 'email-preferences'].includes(mode)) {
+    return location.pathname;
   }
```

Comment:
I think we should try to transition this in future so that we return `location.pathname` by default and then just call out the special cases where we can't do this for some reason. The best outcome of course would be to remove the need for the `/app/` prefix entirely.

Reply:
Agreed